### ISSUE TITLE
🐛 internalError/panic: no entry found for key

### ISSUE THREAD

**Issue by ericprestemon** [biomejs/biome#8473]
### Environment information

```block
CLI:
  Version:                      2.3.9
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.17.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true
  HTML full support enabled:    unset

Workspace:
  Open Documents:               0
```

### What happened?

```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: crates/biome_js_semantic/src/semantic_model/scope.rs:129:39
Thread Name: biome::workspace_worker_0
Message: no entry found for key

triggers-bug.ts internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: no entry found for key
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

Checked 0 files in 2ms. No fixes applied.
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
  
  ℹ Check your biome.json or biome.jsonc to ensure the paths are not ignored by the configuration.
  
  ℹ These paths were provided but ignored:
  
  - triggers-bug.ts
```

[triggers-bug.ts](https://github.com/user-attachments/files/24173934/triggers-bug.ts)

### Expected result

Version 2.3.8 has no issues with the file.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by scriptcoded** [comment]
Was just about to go in and report this and can confirm that I'm seeing the exact same issue referring to the same line (`scope.rs:129:39`).

**Comment by dyc3** [comment]
This is reproducible in the playground: https://biomejs.dev/playground/?code=ZQB4AHAAbwByAHQAIABpAG4AdABlAHIAZgBhAGMAZQAgAFQAeQBwAGUAZABFAHYAZQBuAHQAQwBvAG4AcwB0AHIAdQBjAHQAbwByADwAVAAgAGUAeAB0AGUAbgBkAHMAIABzAHQAcgBpAG4AZwAsACAASQAgAGUAeAB0AGUAbgBkAHMAIABFAHYAZQBuAHQASQBuAGkAdAA%2BACAAewAKAAkAbgBlAHcAIAAoACkAOgAgAEUAdgBlAG4AdAAgACYAIABUAHkAcABlAGQARQB2AGUAbgB0ADwAVAAsACAASQA%2BADsACgB9AAoA (see js console)

**Comment by dyc3** [comment]
I've narrowed it down to `new ()`, removing that text and replacing it with `foo` makes the panic go away. So its got to do with that part specifically

**Comment by ericprestemon** [comment]
[trigger-8473.ts](https://github.com/user-attachments/files/24175951/trigger-8473.ts) is another snippet from our code that also triggers the bug, in case this helps.
